arm-datalake-analytics/account/2016-11-01/swagger/account.json

```diff
@@ -0,0 +1,1580 @@
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Successfully retrieved the specified Azure Storage account information.",
+            "schema": {
+              "$ref": "#/definitions/StorageAccountInfo"
+            }
+          }
+        }
+      },
+      "delete": {
+        "tags": [
+          "Account"
+        ],
+        "operationId": "Account_DeleteStorageAccount",
+        "description": "Updates the specified Data Lake Analytics account to remove an Azure Storage account.",
+        "parameters": [
+          {
+            "name": "resourceGroupName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the Azure resource group that contains the Data Lake Analytics account."
+          },
+          {
+            "name": "accountName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the Data Lake Analytics account from which to remove the Azure Storage account."
+          },
+          {
+            "name": "storageAccountName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the Azure Storage account to remove"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Successfully deleted the specified Azure Storage account from the Azure Data Lake Analytics account."
+          }
+        }
+      },
+      "patch": {
+        "tags": [
+          "Account"
+        ],
+        "operationId": "Account_UpdateStorageAccount",
+        "description": "Updates the Data Lake Analytics account to replace Azure Storage blob account details, such as the access key and/or suffix.",
+        "parameters": [
+          {
+            "name": "resourceGroupName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the Azure resource group that contains the Data Lake Analytics account."
+          },
+          {
+            "name": "accountName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the Data Lake Analytics account to modify storage accounts in"
+          },
+          {
+            "name": "storageAccountName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The Azure Storage account to modify"
+          },
+          {
+            "name": "parameters",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/AddStorageAccountParameters"
```

Comment:
I am talking about the schema of AddStorageAccountParameters. This model has some required properties.

General PATCH semantics says that the body for PATCH should not have required parameters. Since PATCH is essentially  like a git diff. So only the things that one needs to change will be provided everything else remains constant. Over here having a required property in a model means you are forcing the customer to provide a value for a property even if he does not want to change that.